src/PNE/pne_ecrire_jeu_de_donnees_mps.c

```diff
@@ -149,35 +164,48 @@ N : free ??
 fprintf(Flot," N  OBJECTIF\n");
 /* Ecriture de toutes les contraintes */
 for ( Cnt = 0 ; Cnt < NombreDeContraintes ; Cnt++ ) {
+  int NomDeContrainteParDefaut = NomsDesContraintes == NULL || NomsDesContraintes[Cnt] == NULL;
+  MettreAjourNom(NomDeContrainteParDefaut, NomContrainte,NomsDesContraintes[Cnt], Cnt, 'R' );
   if ( Sens[Cnt] == '=' ) {
-    fprintf(Flot," E  R%07d\n",Cnt);
+    fprintf(Flot, " E  %s\n", NomContrainte);
   }
   else if (  Sens[Cnt] == '<' ) {
-    fprintf(Flot," L  R%07d\n",Cnt);
+    fprintf(Flot, " L  %s\n", NomContrainte);
   }
   else if (  Sens[Cnt] == '>' ) {
-    fprintf(Flot," G  R%07d\n",Cnt);
+    fprintf(Flot, " G  %s\n", NomContrainte);
   }
   else {
     fprintf(Flot,"PNE_EcrireJeuDeDonneesMPS : le sens de la contrainte %c ne fait pas partie des sens reconnus\n",
             Sens[Cnt]);
     exit(0);
   } 
+  
 }
 
 /* COLUMNS */
 fprintf(Flot,"COLUMNS\n");
 for ( Var = 0 ; Var < NombreDeVariables ; Var++ ) {
-  if ( CoutLineaire[Var] != 0.0 ) {
-    sprintf(Nombre,"%-.10lf",CoutLineaire[Var]);
+  int NomDeVariableParDefaut = NomsDesVariables == NULL || NomsDesVariables[Var] == NULL;
+  
+  MettreAjourNom(NomDeVariableParDefaut, NomVariable, NomsDesVariables[Var], Var, 'C');
```

Comment:
`NomDeVariableParDefaut` can be computed within `MettreAjourNom` to hide implementation details there.
```suggestion
  MettreAjourNom(NomVariable, NomsDesVariables, Var, 'C');
```
Of course, this remark also applies wherever `MettreAjourNom` is called.